runners/flink/src/test/java/org/apache/beam/runners/flink/website/PipelineOptionsTableGenerator.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink.website;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import org.apache.beam.runners.flink.FlinkPipelineOptions;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.CaseFormat;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+
+/**
+ * A main class which is called by the Gradle generatePipelineOptionsTable* tasks to update the list
+ * of available pipeline options for the Beam website.
+ */
+public class PipelineOptionsTableGenerator {
+
+  private static final List<String> supportedLanguages = ImmutableList.of("java", "python");
+
+  private static class Option {
+    String name;
+    String description;
+    @Nullable String defaultValue;
+
+    public Option(String name, String description, @Nullable String defaultValue) {
+      this.name = name;
+      this.description = description;
+      this.defaultValue = defaultValue;
+    }
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 1) {
+      throw new RuntimeException(
+          "Please specify the language (" + supportedLanguages + ") as the only argument.");
+    }
+    String arg = args[0].toLowerCase();
+    if (!supportedLanguages.contains(arg)) {
+      throw new RuntimeException("The language is not supported: " + arg);
+    }
+    boolean isPython = arg.equals("python");
+
+    printHeader();
+    List<Option> options = extractOptions(isPython);
+    printOptionsTable(options);
```

Comment:
I think using a template library or even a template to generate a simple table is overkill here.

Reply:
I must disagree. Using at least a template and reading it with StringBuilder would simplify the code in this class by separating the view from the generation logic - this alone is a big benefit. Other, previously mentioned points above are still valid too. At the same time, adding the template alone (even only with the StringBuilder reading it) is relatively easy to do.

Right now, future contributors that will need/want to extend/fix this code have to deal with the complexity that otherwise would not be there. It will be therefore harder to generify the solution and adapt it to other runners later and it will probably discourage some to do so.